tomcat-7-valves/src/main/java/org/jboss/openshift/valves/StdoutAccessLogValve.java

```diff
@@ -16,31 +16,1105 @@
+                        synchronized(parent) {
+                            parentCache = parent.getCache(format, loc);
+                        }
+                    }
+                    cache = new Cache(format, loc, parentCache);
+                    formatCache.put(format, cache);
+                }
+            }
+            return cache;
+        }
+
+        public String getFormat(long time) {
+            return cLFCache.getFormatInternal(time);
+        }
+
+        public String getFormat(String format, Locale loc, long time) {
+            return getCache(format, loc).getFormatInternal(time);
+        }
+    }
+
+    private static final DateFormatCache globalDateCache =
+            new DateFormatCache(globalCacheSize, Locale.getDefault(), null);
+
+    private static final ThreadLocal<DateFormatCache> localDateCache =
+            new ThreadLocal<DateFormatCache>() {
+        @Override
+        protected DateFormatCache initialValue() {
+            return new DateFormatCache(localCacheSize, Locale.getDefault(), globalDateCache);
+        }
+    };
+
+    private static final ThreadLocal<Date> localDate = new ThreadLocal<Date>() {
+        @Override
+        protected Date initialValue() {
+            return new Date();
+        }
+    };
+
+    private static enum FormatType {
+        CLF, SEC, MSEC, MSEC_FRAC, SDF
+    }
+
+    private boolean resolveHosts = false;
+
+    private boolean checkExists = false;
+
+    protected String condition = null;
+
+    protected String conditionIf = null;
+
+    protected String localeName = Locale.getDefault().toString();
+
+    protected Locale locale = Locale.getDefault();
+
+    protected String encoding = null;
+
+    protected AccessLogElement[] logElements = null;
+
+    protected boolean requestAttributesEnabled = false;
+
+    public boolean getEnabled() {
+        return enabled;
+    }
+
+    public void setRequestAttributesEnabled(boolean requestAttributesEnabled) {
+        this.requestAttributesEnabled = requestAttributesEnabled;
+    }
+
+    public boolean getRequestAttributesEnabled() {
+        return requestAttributesEnabled;
+    }
+
+    public void setEnabled(boolean enabled) {
+        this.enabled = enabled;
+    }
+
+    @Override
+    public String getInfo() {
+        return (info);
+    }
+
+    public String getPattern() {
+        return (this.pattern);
+    }
+
+    public void setPattern(String pattern) {
+        if (pattern == null) {
+            this.pattern = "";
+        } else if (pattern.equals(Constants.AccessLog.COMMON_ALIAS)) {
+            this.pattern = Constants.AccessLog.COMMON_PATTERN;
+        } else if (pattern.equals(Constants.AccessLog.COMBINED_ALIAS)) {
+            this.pattern = Constants.AccessLog.COMBINED_PATTERN;
+        } else {
+            this.pattern = pattern;
+        }
+        logElements = createLogElements();
+    }
+
+    public boolean isCheckExists() {
+        return checkExists;
```

Comment:
I'm guessing this should be hard coded to false, since we're not logging to file.